lapce-data/src/data.rs

```diff
@@ -275,6 +316,113 @@ impl LapceData {
+                                    }
+                                },
+                                // Second right component isn't a number
+                                Err(_) => {
+                                    // Piece together path components into file path
+                                    let mut str = String::new();
+                                    Self::write_text_with_sep_to(splits.rev(), &mut str, ":").ok()?;
+                                    
+                                    // Convert file path to PathBuf. Ignore final colon
+                                    let left_path = PathBuf::from(
+                                        if str.chars().last().unwrap_or_default() == ':' {
+                                            &str[..str.len() - 1]
+                                        } else {
+                                            &str[..str.len() ]
+                                        }
+                                    );
+                                    
+                                    // rhs1 must be line number and previous components combined must be file name
+                                    if left_path.is_file() {
+                                        return Some((
+                                            left_path,
+                                            Some( LineCol { line: first_rhs_num, column: 1 } ),
+                                        ))
+                                    }
+                                },
+                            }
+                        }
+                    },
+                    // First right component isn't a number
+                    Err(_) => {
+                        // If the first right component isn't a number, then the enire path has to refer to a file name
+                        if path.is_file() {
+                            return Some((
+                                path.to_path_buf(),
+                                Some( LineCol { line: 1, column: 1 } ),
+                        ));
+                        }
+                    }
+                }
+            }
+        // Check if the path refers to a file if the path isn't UTF-8 valid
+        } else if path.is_file() {
+            return Some((
+                path.to_path_buf(), 
+                Some( LineCol { line: 1, column: 1 } )
+            ))
+        }
+        // Ignore the path if it doesn't refer to a file
+        None
+    }
+    
+    pub fn write_text_with_sep_to<I, T, Buf>(
+        mut iter: I, 
+        buf: &mut Buf, 
+        sep: T,
+    ) -> fmt::Result 
+    where
+        Buf: fmt::Write,
+        I: Iterator<Item = T>,
+        T: AsRef<str>, 
+        
+    {
+        if let Some(str) = iter.next() {
+            buf.write_str(str.as_ref())?;
+            buf.write_str(sep.as_ref())?;
+            Self::write_text_with_sep_to(iter, buf, sep)?
+        }
+        Result::Ok(())
+    }
```

Comment:
And further. This function has nothing to do with `LapceData`. Please move it inside `parse_path_line_column` function, and then you can probably rewrite it simpler (since generics and error checking are not needed then):
```rust
fn write_text_with_sep_to<'a, I>(mut iter: I, buf: &mut String, sep: &str)
where
    I:  Iterator<Item = &'a str>,
{
    if let Some(str) = iter.next() {
        buf.push_str(str);
        buf.push_str(sep);
        write_text_with_sep_to(iter, buf, sep);
    }
}
```